### PR TITLE
[GCI] Add send to self feature for custom emails

### LINKED ISSUE #425: Admin: Send Custom Mail Preview to Self
When admin creates a mail and would like to send it, it would be nicer if we could test send a mail to admin first. This is necessary because sometimes mails turn out unexpected and doesn't match the preview mail shown. 

**Describe the solution you'd like**
Along with **edit** and **send**, a **send to self** option is desirable. 
page: `http://localhost:3000/custom_mails/1` 
image

**Are you working on this? (Yes/No)**
No

### PR DESCRIPTION
**Pull request by junron**
Fixes #425 

#### Describe the changes you have made in this pr -
Added link to send email to self.
[MailCatcher](https://mailcatcher.me/) was used to view mails in development.

### Demonstration video:
demo

[GCI task](https://codein.withgoogle.com/dashboard/task-instances/6384327542177792/)

### CODE REVIEW

**Comment by junron** [comment]
Ughh the GitHub GIF quality is really bad

**Comment by junron** [comment]
@shubhankarsharma00 Please review

**Comment by tachyons** [comment]
cc @tonyvince

**Comment by shubhankarsharma00** [comment]
> @shubhankarsharma00 Please review

let's wait for @satu0king 's review.

**Comment by junron** [comment]
@sat0king Please review

**Comment by junron** [comment]
@satu0king please review

File: app/controllers/custom_mails_controller.rb [hunk 1]
```diff
@@ -54,7 +54,17 @@ def send_mail
     render html: "The mails were queued for sending!"
   end
 
+  def send_mail_self
+    @mail.sent = false
```
Comment (ankitkataria) [inline app/controllers/custom_mails_controller.rb]: Why is this required?

Comment (junron) [reply app/controllers/custom_mails_controller.rb]: @ankitkataria `self_mail_self` is the route handler. I don't think it is suitable to reuse the batch email handler because it's a different route. 

As for the `@mail.sent = false`, this sending email to self is more of a preview.

I hope this addresses your concerns, and please ask more questions if m clarification is inadequate.

Thank you!

Comment (ankitkataria) [reply app/controllers/custom_mails_controller.rb]: > As for the @mail.sent = false, sending email to self is more of a preview, rather than sending the email out to the public.

It will be false be default. We can remove this from here.

Comment (junron) [reply app/controllers/custom_mails_controller.rb]: Ok, I will remove that line.

**Comment by junron** [comment]
@ankitkataria Fixed, please review

**Comment by junron** [comment]
@ankitkataria @shubhankarsharma00 please take a look

**Comment by junron** [comment]
@ankitkataria @shubhankarsharma00 Please review

**Comment by shubhankarsharma00** [comment]
@junron how do you start the mailcatcher on our particular IP?

**Comment by junron** [comment]
@shubhankarsharma00 Edit development.rb to point to localhost, then ssh tunnel localhost 1080 on the server to my computer.

**Comment by shubhankarsharma00** [comment]
So basically you changed this?
https://github.com/CircuitVerse/CircuitVerse/blob/d64e1a0c3db31871160b4af3f98946a7ae934242/config/environments/development.rb#L61-L63

**Comment by junron** [comment]
Yeah, I changed it to:
```
config.action_mailer.smtp_settings = {
  :address              => 'localhost',
  :port                 => 1025
}
```
I didn't think it was necessary to include that as MailCatcher was not part of the task.

**Comment by shubhankarsharma00** [comment]
It is cool but the issue is for production part and changing SMTP settings for production is not desirable. Can we do it in some other way ?

File: app/controllers/custom_mails_controller.rb [hunk 2]
```diff
@@ -54,7 +54,16 @@ def send_mail
     render html: "The mails were queued for sending!"
   end
 
+  def send_mail_self
+    @mail.save
```
Comment (ankitkataria) [inline app/controllers/custom_mails_controller.rb]: This will also not be required @junron

**Comment by junron** [comment]
MailCatcher does not affect the operation of the application in any way. If you run the code in this pull request in production, it will still use your SMTP servers. This applies to development as well, given a valid username and password is provided

File: app/controllers/custom_mails_controller.rb (hunk 2 above)
Comment (junron) [reply app/controllers/custom_mails_controller.rb]: Fixed @ankitkataria

**Comment by ankitkataria** [comment]
Nice work so far @junron ! :rocket: 
It is possible to add tests for the new route added? So that the test coverage is taken care of?
Refer `custom_mails_controller_spec.rb`. :wink:

**Comment by junron** [comment]
No problem! I'll also fix the hound violations

**Comment by tachyons** [comment]
Ignore the hound violations for now. Need to configure that properly

**Comment by shubhankarsharma00** [comment]
@junron ah yeah thanks for clarifying, pardon the query before. Great work.

**Comment by junron** [comment]
@ankitkataria Added tests
@shubhankarsharma00 No problem, feel free to ask questions
